### PR TITLE
Support for OpenSSL 1.1

### PR DESCRIPTION
**Pull request by postwait**

### CODE REVIEW

File: src/eventer/eventer_SSL_fd_opset.c [hunk 1]
```diff
@@ -1269,22 +1330,36 @@ int eventer_ssl_config(const char *key, const char *value) {
   }
   return 1;
 }
+
+static int32_t initialized = 0;
```
Comment (cody271) [inline src/eventer/eventer_SSL_fd_opset.c]: Needs `volatile` here? I think `eventer_ssl_init()` could be called from different threads..

File: src/modules/lua_mtev_crypto.c [hunk 2]
```diff
@@ -84,7 +90,7 @@ mtev_lua_crypto_x509_index_func(lua_State *L) {
   cert = *((X509 **)udata);
   if(!strcmp(k, "signature_algorithm")) {
     int nid;
-    nid = OBJ_obj2nid(cert->sig_alg->algorithm);
+    nid = X509_get_signature_nid(cert);
```
Comment (cody271) [inline src/modules/lua_mtev_crypto.c]: What does this change here do?

File: src/modules/lua_mtev_crypto.c [hunk 3]
```diff
@@ -576,7 +616,6 @@ static int __attribute__((unused)) mtev_lua_crypto_bn_##name(lua_State *L) { \
 static int mtev_lua_crypto_bn_is_negative(lua_State *L) {
   BN_METH_DECL_INT(0);
   lua_pushboolean(L, BN_is_negative(bn));
-  lua_pushboolean(L, bn->neg ? 1 : 0);
   return 2;
```
Comment (cody271) [inline src/modules/lua_mtev_crypto.c]:
```suggestion
   return 1;
```

File: src/utils/mtev_stacktrace.c [hunk 4]
```diff
@@ -426,6 +426,11 @@ mtev_dwarf_walk_map(void (*f)(const char *, uintptr_t)) {
 #else
 #endif
 }
+#else
```
Comment (cody271) [inline src/utils/mtev_stacktrace.c]: How are these changes related to OpenSSL ..?

File: src/eventer/eventer_SSL_fd_opset.c (hunk 1 above)
Comment (postwait) [reply src/eventer/eventer_SSL_fd_opset.c]: It can't really.. We call this once, this is just protection in case a user is dumb and calls it again.

File: src/modules/lua_mtev_crypto.c (hunk 2 above)
Comment (postwait) [reply src/modules/lua_mtev_crypto.c]: It doesn't change anything, but cert is opaque in openssl 1.1

File: src/utils/mtev_stacktrace.c (hunk 4 above)
Comment (postwait) [reply src/utils/mtev_stacktrace.c]: They aren't, I'll separate them. (same PR, but separate commit)
